src/shared/libmount-util.h

```diff
@@ -3,20 +3,29 @@
 
 #include <stdio.h>
 
+#if HAVE_MOUNT
 /* This needs to be after sys/mount.h */
 #include <libmount.h>
+#endif
 
 #include "macro.h"
 
+#if HAVE_MOUNT
 DEFINE_TRIVIAL_CLEANUP_FUNC(struct libmnt_table*, mnt_free_table);
 DEFINE_TRIVIAL_CLEANUP_FUNC(struct libmnt_iter*, mnt_free_iter);
+#else
+/* "Bogus" forward declarations for the libmount_parse parameters below */
```

Comment:
@evverx I added several other guards to other files, as the build was still failing. Also, I had to add the two following forward declarations, as the `libmount_parse` functions needed them. Not sure if it's an ideal solution (but it works!) so any feedback is welcome :-)

Reply:
@mrc0mmand not that there is anything inherently wrong with that commit but it's worth mentioning that I haven't put a lot of engineering into #12608 and the commit message saying that "It's just duct tape and popsicle sticks for now" reflects that hopefully :-) I think it would be better to think about how to properly skip libmount and libcap and then try to implement that in a separate PR.

Reply:
This and the rest of the commits are definitely a subject to change, I just didn't want to *ruin* your duct tape & popsicle sticks commit by adding a cheap superglue :-)